x-pack/plugins/security_solution/public/common/utils/use_set_field_value_cb.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FormHook } from '@kbn/es-ui-shared-plugin/static/forms/hook_form_lib';
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+export const useSetFieldValueWithCallback = ({
+  field,
+  setFieldValue,
+  value,
+}: {
+  field: string;
+  value: unknown;
+  setFieldValue: FormHook['setFieldValue'];
+}) => {
+  const isWaitingRef = useRef(false);
+  const valueRef = useRef();
+  const [callback, setCallback] = useState<() => void>(() => null);
+
+  useEffect(() => {
+    if (isWaitingRef.current && value === valueRef.current && typeof callback === 'function') {
```

Comment:
callback will always be a function according to the default? Did yo uwant to default it to just be `null` on line 22?

Reply:
(thats what i get for copy/pasta from SO) :D